tools/autograd/gen_variable_type.py

```diff
@@ -120,47 +399,128 @@ def unpack_args(option):
+            option['tensor_args'] = [arg['name'] for arg in option['arguments']
+                                     if arg['dynamic_type'] == 'Tensor']
+        if return_type == 'Scalar':
+            option['return_value'] = 'Scalar(output)'
+        else:
+            option['return_value'] = 'Tensor(std::move(output))'
+
         option['type_definition_body'] = emit_body(option)
 
         type_declarations.append(METHOD_DECLARATION.substitute(option))
-        type_definitions.append(METHOD_DEFINITION.substitute(option))
-
-    def emit_body(option):
-        body = []
-        body += unpack_args(option)
-
-        if option['return_type'] in FALLTHROUGH_RETURN_TYPES:
-            body.extend(METHOD_DEFINITION_FALLTHROUGH.substitute(option).split('\n'))
-            return body
-
-        return METHOD_DEFINITION_NYI.substitute(option)
+        if option['name'] != 'resize_':
+            type_definitions.append(METHOD_DEFINITION.substitute(option))
 
     for function in aten_declarations:
         process_function(function)
 
 
-def create_python_bindings(top_env, aten_decls, derivatives):
+def create_python_bindings(top_env, python_functions):
     """python_variable_methods.cpp
 
     Generates Python bindings to Variable methods
     """
+    py_methods = top_env['py_methods']
+    py_method_defs = top_env['py_method_defs']
+    py_method_dispatch = top_env['py_method_dispatch']
 
-    # py_methods = top_env['py_methods']
-    # py_method_defs = top_env['py_method_defs']
-
-    def process_option(option):
-        # TODO: generate Python bindings
-        pass
+    unpack_methods = {
+        'int64_t': 'toInt64',
+        'bool': 'toBool'
+    }
 
-    for option in aten_decls:
-        process_option(option)
+    def args_without_self(args):
+        return [arg for arg in args if arg['name'] != 'self']
+
+    def emit_dispatch(i, option):
+        env = {}
+
+        args = []
+        formal_args = ['Tensor & self']
+        python_params = args_without_self(option['python_arguments'])
+        for arg_idx, arg in enumerate(python_params):
+            unpack = unpack_methods.get(arg['type'], arg['type'].lower())
+            args.append('r.{}({})'.format(unpack, arg_idx))
+            dispatch_type = arg['type']
+            dispatch_type = dispatch_type.replace('Tensor', 'const Tensor &')
+            formal_args.append('{} {}'.format(dispatch_type, arg['name']))
+
+        env['i'] = i
+        env['dispatch_args'] = [arg for arg in option['call_args'] if arg != 'self']
+        env['args_with_self'] = ['self_'] + args
+        env['AutoNoGIL'] = 'AutoNoGIL no_gil;'
+        env['AutoGPU'] = 'AutoGPU auto_gpu(self);'
+        env['formal_args'] = formal_args
+        env['cond'] = 'if' if i == 0 else '} else if'
+        env = nested_dict(env, option)
+        py_method_dispatch.append(PY_VARIABLE_DISPATCH.substitute(env))
+        return PY_VARIABLE_CASE.substitute(env)
```

Comment:
Is there any reason why we want to keep these dispatch functions separate from the place where they are used?